Consider super event when resolving enrolment status for an event

nikomakela opened the pull request:
HH-449.

1. Use super event to populate registration data for sub event in cases when sub event has registration set to null, but has enrolment data given in event details root.
2. Fix super event id parsing from internal id -link when dealing with course typed events.
3. Remove `getLinkArrowLabel` component as redundant.
4. Move eventUtils' tests from app to components-package.

nikomakela committed:
fix: event enrolment status handling

HH-449.

`getEnrolmentStatus` returned full as a status, when the enrolling had
already ended. The enrolment times should be checked before the
remaining attendee status.

`getEnrolmentStatus` should also support checking the registration data
from the super event, because in LinkedEvents, the enrolling logic seems
to work so, that it's using super event's registration, when subevent
has set `event.registration` -field to `null`, but super event has it
populated.

nikomakela committed:
chore: remove getLinkArrowLabel as redundant

HH-449.

getLinkArrowLabel is no longer used and it had a bad design from the
beginning.

nikomakela committed:
feat: use super event registration data to resolve enrolment status

HH-449.

When dealing with a subevent, it might be that only the super event has
the registration enabled. In these cases the registration should be
inherited to the subevent. So when resolving the enrolment status of an
event, we should also consider the superevent.

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr884.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr884.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr884.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr884.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
test: move event utils test from events-app to components package

HH-449.

The eventUtils tests had been left in events-helsinki app, when they
should have been in packages/components at this point.

nikomakela committed:
fix: getEventIdsFromUrls should work with course typed events too

HH-449.

This change fixes super event queries in course typed event details
page.

getEventIdsFromUrls was hard fixed to events, but the course type events
have "course" in the URL instead.

Because the super event link in event details documents has contained
links that contains "/course" in the link instead of "/event", the super
event query has not been working.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr884.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela commented:
Test result: There are some render issues still. Superevent query is a lazy query, but it is not indicated to the user anyhow. For some reason, it's not always updated (rerendered) properly and a page reload is needed.

- [ ] Fix superevent data (re)rendering issues

nikomakela committed:
fix: useIsEnrolmentOpen should support super event in offer button

HH-449.

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr884.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix: upgrade HCRC-lib to fix duplicate fragment

HH-449.

nikomakela committed:
fix: super event handling in event hero enrolment features

HH-449.

Sometimes, for the subevents, the LinkedEvents handles the registration
so that it inherits the registration from super event. In these cases,
the super event registration data should be used to decide what
enrolment status text is shown or whether the offer button is shown or
not.

This commit
- fixes the useSuperEventLazyLoad by making it more reliable
- adds a skeleton loader to LargeEventCard and it's enrolment status
indicator
- adds a skeleton loader to event hero enrolment features when the super
event is still loading.
- refactors super event handling to be more reliable
- adds unit tests

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr884.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders commented on packages/components/src/utils/__tests__/getEventEnrolmentStatus.test.ts:
```diff
@@ -0,0 +1,275 @@
+  return date.toISOString();
+};
+
+const MOCK_NOW = '2023-10-26T12:00:00.000Z';
```
MOCK_NOW could be used in getRelativeDate.

karisal-anders reviewed (approved):
Some minor things, left some comments, please take a look at them.

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr884.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr884.dev.hel.ninja** :rocket::rocket::rocket:
